connectors/sync_job_runner.py

```diff
@@ -238,11 +240,20 @@ async def sync_starts(self):
     async def prepare_docs(self):
         logger.debug(f"Using pipeline {self.sync_job.pipeline}")
 
-        async for doc, lazy_download in self.data_provider.get_docs(
-            filtering=self.sync_job.filtering
-        ):
+        if self.sync_job.job_type == JobType.FULL:
+            generator = self.data_provider.get_docs(filtering=self.sync_job.filtering)
+        elif self.sync_job.job_type == JobType.INCREMENTAL:
+            generator = self.data_provider.get_docs_incrementally(
+                sync_cursor=self.connector.sync_cursor,
+                filtering=self.sync_job.filtering,
+            )
+        else:
+            raise UnsupportedJobType
+
+        async for doc, lazy_download, *operation in generator:
             doc_id = str(doc.get("_id", ""))
             doc_id_size = len(doc_id.encode(UTF_8))
+            operation = operation[0] if operation else "index"
```

Comment:
I would suggest we also use constants for these operations like `INDEX = index` or `OP_INDEX = index` here, so renaming/refactoring will be easier + spelling mistakes are less likely. I guess we would need to move those constants then to avoid circular imports.